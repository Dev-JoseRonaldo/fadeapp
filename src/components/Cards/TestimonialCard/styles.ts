export const testimonialCardStyle = `
flex
flex-col
md:flex-row
lg:w-[54.813rem]
w-max
relative
justify-center
ml-7
lg:ml-0
`
export const nameStyle = `
md:absolute
w-[18.75rem]
md:left-[1.875rem]
md:top-[3.438rem]
text-white
break-words
overflow-hidden
max-h-8
mb-4
md:mb-0
text-center
font-medium
md:text-left
text-2xl
box-border
`

export const messageStyle = `
text-white
break-words
overflow-hidden
max-h-[160px]
text-center
mb-8
md:mb-0
md:text-left
relative
text-base
md:text-lg
md:top-[6.875rem]
md:left-[-6.875rem]
w-[18.75rem]
`

export const photoContainerStyle = `
relative
box-border
flex
justify-center
items-center
w-[18.75rem]
h-[17.5rem]
rounded-10
md:left-[-2.125rem]
shadow-[0.5rem_0.5rem_0rem_0rem_rgba(33,231,182)]


after:content-[' ']
after:block
after:absolute
after:h-[0.25rem]
after:w-[1.875rem]
after:top-[11.25rem]
after:right-[-1.875rem]
md:after:w-[10.75rem]
md:after:top-[5.625rem]
md:after:right-[-10.75rem]
after:bg-primary


before:content-[' ']
before:block
before:absolute
md:before:h-[0.25rem]
md:before:w-[25.313rem]
md:before:top-[5.625rem]
md:before:left-[-25.313rem]
before:bg-primary
`
