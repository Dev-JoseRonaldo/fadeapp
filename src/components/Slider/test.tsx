import { render, screen } from '@testing-library/react'

import { testimonialUsersMock } from 'components/Cards/TestimonialCard/mock'

const { getAllByTestId } = screen

import { Slider } from '.'

jest.mock('components/Cards/TestimonialCard', () => ({
  __esModule: true,
  TestimonialCard: function Mock({ children }: { children: React.ReactNode }) {
    return <div data-testid="Mock TestimonialCard">{children}</div>
  },
}))

describe('<Slider />', () => {
  it('should render slider', () => {
    render(<Slider data={testimonialUsersMock} />)

    const testimonialCards = getAllByTestId('Mock TestimonialCard')

    expect(testimonialCards).toHaveLength(testimonialUsersMock.length)
  })

  it('should match snapshot', () => {
    const { container } = render(<Slider data={testimonialUsersMock} />)

    expect(container.firstChild).toMatchSnapshot()
  })
})
