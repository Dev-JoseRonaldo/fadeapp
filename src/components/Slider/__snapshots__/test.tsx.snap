// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Slider /> should match snapshot 1`] = `
<div
  class="
flex
"
>
  <div
    class="
hidden cursor-pointer lg:flex mr-[4.25rem]
"
  >
    <img
      alt="Seta maior"
      src="/Vector2.svg"
    />
    <img
      alt="Seta menor"
      src="/Vector.svg"
    />
  </div>
  <div
    class="
 overflow-hidden
 max-w-[1200px]
 flex
 md:justify-around
 bg-[#212948]
 h-[36.25rem]  
 py-[2rem]
 md:py-[0rem]
 md:h-[29.25rem]
 md:items-center
"
  >
    <div
      class="
m-0 mr-[4rem] w-full md:mr-0 md:pl-[4rem] 
"
    >
      <div
        class="
ease flex  w-full transition items-end
"
        draggable="true"
        style="transform: translateX(0%);"
      >
        <div
          class="inline-block h-full"
        >
          <div
            data-testid="Mock TestimonialCard"
          />
        </div>
        <div
          class="inline-block h-full"
        >
          <div
            data-testid="Mock TestimonialCard"
          />
        </div>
        <div
          class="inline-block h-full"
        >
          <div
            data-testid="Mock TestimonialCard"
          />
        </div>
        <div
          class="inline-block h-full"
        >
          <div
            data-testid="Mock TestimonialCard"
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
